src/payment/payment.service.ts

```diff
@@ -0,0 +1,142 @@
+        });
+        await this.prismaService.payment.update({
+          data: { status: Status.COMPLETED },
+          where: { id: await this.getPaymentId(confrimPaymentDto.bookingId) },
+        });
+        return this.setMessage(
+          true,
+          await translateMessage(this.i18n, MessagesTypes.BOOKING_PAYMENT_SUCCESS)
+        );
+      case 'processing':
+        return this.setMessage(
+          true,
+          await translateMessage(this.i18n, MessagesTypes.BOOKING_PAYMENT_PROCESSING)
+        );
+      case 'requires_payment_method':
+        return this.setMessage(
+          false,
+          await translateMessage(this.i18n, MessagesTypes.BOOKING_PAYMENT_REQUEIRED_METHOD)
+        );
+      default:
+        return this.setMessage(
+          false,
+          await translateMessage(this.i18n, MessagesTypes.BOOKING_PAYMENT_FAILED)
+        );
+    }
+  }
+
+  async processCashPayment(createPaymentDto: CreatePaymentDto) {
+    try {
+      const totalAmount = await this.getTotalAmount(createPaymentDto.bookingId);
+      await this.prismaService.payment.create({
+        data: {
+          type: createPaymentDto.paymentOption,
+          totalAmount,
+          status: Status.COMPLETED,
+        },
+      });
+
+      await this.prismaService.booking.updateMany({
+        where: { id: createPaymentDto.bookingId, status: Status.PENDING },
+        data: { status: Status.COMPLETED },
+      });
+
+      return {
+        success: true,
+        message: await translateMessage(this.i18n, MessagesTypes.BOOKING_PAYMENT_SUCCESS),
+      };
+    } catch (error) {
+      throw new GlobalException(ErrorsTypes.PAYMENT_FAILED_TO_PROCESS, error.message);
+    }
+  }
+
+  async getTotalAmount(bookingId: string) {
+    const bookingDetails = await this.prismaService.booking.findFirst({
+      where: { id: bookingId, status: Status.PENDING },
+    });
+    if (bookingDetails == null) {
+      throw new NotFoundException(ErrorsTypes.NOT_FOUND_BOOKING);
+    }
+
+    const accommodationDetails = await this.prismaService.accommodation.findFirst({
+      where: { id: bookingDetails.accommodationId },
+    });
+    if (accommodationDetails == null) {
+      throw new NotFoundException(ErrorsTypes.NOT_FOUND_ACCOMMODATION);
+    }
+    const startDay = dayjs(bookingDetails.startDate);
+    const endDay = dayjs(bookingDetails.endDate);
+    const amountOfDays = endDay.diff(startDay, 'day');
+    return accommodationDetails.price * amountOfDays;
+  }
+
+  async setMessage(success: boolean, message: string) {
+    return { success, message };
+  }
+
+  async getPaymentId(bookingId: string) {
+    const bookingDetails = await this.prismaService.booking.findUnique({
+      where: { id: bookingId },
+    });
+    if (bookingDetails == null) {
+      throw new NotFoundException(ErrorsTypes.NOT_FOUND_BOOKING);
+    }
+    const paymentDetails = await this.prismaService.payment.findFirst({
+      where: { id: bookingDetails.paymentId! },
```

Comment:
why do you use '!' here? don't booking details has type?

Reply:
if I do not use , I am getting that error and I could get why it is happening: 

Type 'string | null' is not assignable to type 'string | StringFilter<"Payment"> | undefined'.
  Type 'null' is not assignable to type 'string | StringFilter<"Payment"> | undefined'.ts(2322)
(property) id?: string | Prisma.StringFilter<"Payment"> | undefined

if you have any idea , can you explain here?